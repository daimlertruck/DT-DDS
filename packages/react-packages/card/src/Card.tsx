import { BaseProps } from '@dt-dds/react-core';
import { Typography } from '@dt-dds/react-typography';

import { CardStyled, CardTitleStyled, CardDetailStyled } from './Card.styled';

export const Card = ({ children, dataTestId }: BaseProps) => (
  <CardStyled data-testid={dataTestId ?? 'card'}>{children}</CardStyled>
);

Card.Title = ({ children }: BaseProps) => {
  return (
    <CardTitleStyled>
      <Typography element='h2' fontStyles='h4'>
        {children}
      </Typography>
    </CardTitleStyled>
  );
};

Card.Detail = ({ children }: BaseProps) => {
  return <CardDetailStyled>{children}</CardDetailStyled>;
};
