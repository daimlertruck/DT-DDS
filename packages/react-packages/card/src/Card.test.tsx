import { withProviders } from '@dt-ui/react-core';
import { render } from '@testing-library/react';

import { Card } from './Card';

describe('<Card /> component', () => {
  const ProvidedCard = withProviders(Card);
  it('renders div html element with the correct items', () => {
    const { container } = render(
      <ProvidedCard>
        <Card.Title>Card Title</Card.Title>
        <Card.Detail>Card Description</Card.Detail>
      </ProvidedCard>
    );

    expect(container).toMatchSnapshot();
  });
});
