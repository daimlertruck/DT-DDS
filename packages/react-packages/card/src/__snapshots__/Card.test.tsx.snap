// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> component renders div html element with the correct items 1`] = `
.emotion-0 {
  margin: 0;
  background-color: #FFFFFF;
  width: 100%;
  max-width: 530px;
  height: 100%;
  padding: 47px 24px 52px 24px;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  border-radius: 3px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-0::after {
  content: '';
  background-color: #000000;
  border-radius: 4px;
  width: 40px;
  height: 5px;
  margin-top: 14px;
}

.emotion-0:hover {
  box-shadow: 1px 4px 4px rgba(0, 0, 0, 0.2);
}

.emotion-2 {
  width: 100%;
  display: -webkit-box;
  -webkit-line-clamp: 1;
  -webkit-box-orient: vertical;
  text-overflow: ellipsis;
  overflow: hidden;
}

.emotion-4 {
  font-weight: 700;
  font-size: 1.25rem;
  line-height: 1.5rem;
  letter-spacing: normal;
  margin: 0;
  text-transform: none;
  color: #292929;
}

@media only screen and (min-width:1024px) {
  .emotion-4 {
    font-size: 1.5rem;
    line-height: 1.75rem;
  }
}

.emotion-5 {
  margin-top: 12px;
  width: 100%;
  height: auto;
  max-height: 48px;
  display: -webkit-box;
  -webkit-line-clamp: 2;
  -webkit-box-orient: vertical;
  text-overflow: ellipsis;
  overflow: hidden;
}

<div>
  <div
    class="emotion-0 emotion-1"
    data-testid="card"
  >
    <div
      class="emotion-2 emotion-3"
    >
      <h2
        class="emotion-4"
        color="grey_90"
        data-testid="typography"
      >
        Card Title
      </h2>
    </div>
    <div
      class="emotion-5 emotion-6"
    >
      Card Description
    </div>
  </div>
</div>
`;
